star-dash/star-dash/GameBridge/SyncModule/ObjectModule.swift

```diff
@@ -0,0 +1,31 @@
+import CoreGraphics
+import SDPhysicsEngine
+
+class ObjectModule: SyncModule {
+    let entityManager: EntitySyncInterface
+
+    init(entityManager: EntitySyncInterface) {
+        self.entityManager = entityManager
+    }
+
+    func sync(entity: Entity, from object: SDObject) {
+        guard let positionComponent = entityManager.component(ofType: PositionComponent.self, of: entity.id) else {
+            return
+        }
+
+        positionComponent.setPosition(position: object.position)
+        positionComponent.setRotation(rotation: Float(object.rotation))
+    }
+
+    func sync(object: SDObject, from entity: Entity) {
+        guard let positionComponent = entityManager.component(ofType: PositionComponent.self, of: entity.id) else {
+            return
+        }
+
+        object.position = positionComponent.position
+        object.rotation = CGFloat(positionComponent.rotation)
```

Comment:
should we just be consistent in `PositionComponent` and use `CGFloat` for the rotation?